File: 26 - Remove Duplicates From Sorted Array/solution.py
from typing import List

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        n = len(nums)
        t = 1
        for r in range(1, n):
            # Find where value changes and update the tail element
            if nums[r] != nums[r-1]:
                nums[t] = nums[r]
                t += 1
        return t

File: 26 - Remove Duplicates From Sorted Array/Solution.java
class Solution
{
    public int removeDuplicates(int[] nums)
    {
        int tail = 1;
        for (int i=1; i<nums.length; i++)
        {
            if (nums[i-1] != nums[i])
            {
                nums[tail++] = nums[i];
            }
        }
        return tail;
    }
}